pkg/cli-runtime/testing/validatable_table.go

```diff
@@ -60,24 +72,31 @@ func (ts ValidatableTestSuite) Run(t *testing.T) {
 	}
 	if len(focused) != 0 {
 		for _, tc := range focused {
-			tc.Run(t)
+			tc.Run(t, ctx)
```

Comment:
To @shaheerkootteeri  earlier point, context should not shared across all tests. Update func signature to `Run(t)`

Reply:
then how do i create tests for validate?

> so when I was trying to create the validatable cases for this, every time the case got to the this part
> 
> ```
> if opts.UpdateStrategy != "" && cli.CommandFromContext(ctx).Flags().Changed(cli.StripDash(flags.UpdateStrategyFlagName))
> ```
> 
> Since there was just an empty context, there were no flags and I was getting a `nil` error . I needed to add the context in order to set the command in the test during the `preprare` stage

Reply:
ctx := context.Background()
create this in the Run func and pass it to validatable

Reply:
But that's what I'm currently doing

Reply:
Please [check](https://github.com/vmware-tanzu/apps-cli-plugin/blob/6c161dd35c2388739993cea98a767a91c768bb40/pkg/cli-runtime/testing/validatable_table.go#L66)

Reply:
updated as mentioned in another comment